frontend/app/routes/api.switch-language.ts

```diff
@@ -0,0 +1,63 @@
+import { type ActionFunctionArgs, json } from '@remix-run/node';
+
+import { z } from 'zod';
+
+import { createLangCookie } from '~/utils/locale-utils.server';
+import { getLogger } from '~/utils/logging.server';
+
+const log = getLogger('routes/api.switch-language');
+
+const switchLanguageDataSchema = z.object({
+  language: z.enum(['en', 'fr']),
+});
+
+export type SwitchLanguageData = z.infer<typeof switchLanguageDataSchema>;
+
+/**
+ * A loader function that always throws a 404 Not Found response.
+ */
+export async function loader() {
+  throw json('404 Not Found', { status: 404 });
+}
+
+/**
+ * Action function to handle the language switch request.
+ *
+ * @param {ActionFunctionArgs} args - The action function arguments.
+ * @returns {Promise<Response>} A promise that resolves to the language switch response.
+ *
+ * @example
+ * const request = //... create a request object
+ * const response = await action({ request });
+ * console.log(response.status); // Output: 204
+ *
+ * @param {ActionFunctionArgs} args - The action function arguments.
+ */
+export async function action({ request }: ActionFunctionArgs) {
+  if (request.method !== 'POST') {
+    log.debug(`Status 405: Invalid request method ${request.method}`);
```

Comment:
Should this be logged as a warning, maybe?

Reply:
Fixed